### ISSUE TITLE
[CocoaPods 0.36.0.rc.1 issues] Renaming Dynamic Library Install Name

### ISSUE THREAD

**Issue by steilerDev** [AgileBits/onepassword-app-extension#180]
I want to integrate the 1Password extension to my Swift application. I am using CocoaPods to integrate your extension into my application. Concluding I needed to use an Objective-C bridging header. 

I had big troubles integrating the extension using this approach, because I was not able to import the required header files using the intuitive command `#import <1PasswordExtension/OnePasswordExtension.h`, because xCode was unable to find the header file.

The configuration provided by CocoaPods the dynamic library install name is _1PasswordExtension, concluding I need to import the extension using the very non-intuitive  `#import <_1PasswordExtension/OnePasswordExtension.h` command. 

It would be great to either include a notice about this non-intuitive behaviour or change the name within the next release.

**Comment by radazzouz** [comment]
Hi @steilerDev,

Thanks for taking the time to write in :+1: 

The library is be named `1PasswordExtension` and not `_1PasswordExtension` in both our [podsepc](https://github.com/AgileBits/onepassword-app-extension/blob/master/1PasswordExtension.podspec) and and in [CocoaPods/Specs](https://github.com/CocoaPods/Specs/commit/d29775f59ad34969aeb2bb4f0d2eefd34709317d).

I am not sure I fully understand the issue, because I cannot reproduce it. Could it be an Xcode or a Swift issue? Could you please send us a sample project so we can try and figure it out on our end?

I look forward to your reply :smile: 

Enjoy your weekend!

**Comment by tapi** [comment]
@steilerDev what version of cocoapods are you using? I did up the pods pec originally but haven't been working with any of the recent betas. I know 0.36 adds framework support but I'm not sure how they are named.

**Comment by steilerDev** [comment]
@radazzouz: I've now pushed my current project:
You can see the usage of your extension in my [LoginViewController](https://github.com/steilerDev/myVerein/blob/master/Code/iOS%20Client/myVerein/myVerein/LoginViewController.swift) and within my [Objective-C bridging header](https://github.com/steilerDev/myVerein/blob/master/Code/iOS%20Client/myVerein/myVerein/myVerein-Bridging-Header.h). Notice that I had to use `import _1PasswordExtension`, respectively `# import <_1PasswordExtension/OnePasswordExtension.h>` to use the extension.

And yes it might be a xCode or Swift issue, I just don't know

@tapi I am using 0.36.0.rc.1 since I need swift framework support

**Comment by radazzouz** [comment]
Hi @steilerDev,

I created a [sample project](https://com-agilebits-users.s3.amazonaws.com/rad/ACME%20CocoaPods.zip) in Objective-C based on **ACME**. I installed CocoaPods `0.36.0.rc.1` and set it up. Please refer to the `Podfile` in the attached sample project.

Note that I used `#import "OnePasswordExtension.h"` and not `#import <_1PasswordExtension/OnePasswordExtension.h>` or `#import <1PasswordExtension/OnePasswordExtension.h>`.

Can you please modify the sample project and send it back to me so I can reproduce the issue here. This way we can clearly determine whether it's an issue that we can fix ourselves or whether it's an Xcode, Swift or CocoaPods issue.

Cheers!

**Comment by tapi** [comment]
Sorry @radazzouz  I did look into this. Theres a blog post [here](http://blog.cocoapods.org/Pod-Authors-Guide-to-CocoaPods-Frameworks/)

===TL;DR 
I think `module_name = '1PasswordExtension'` should be added to the Podfile. Haven't/Don't really have time to test @steilerDev are you able to make this change and see if that cleans up the import?

**Comment by radazzouz** [comment]
@tapi where should this change be made? Should we make it in the `1PasswordExtension.podspec` in this repo or should it be made by developers who are using CocoaPods in their project's `Podfile`?

**Comment by tapi** [comment]
Should be made in the podspec. The actual name can be whatever you want

**Comment by radazzouz** [comment]
Now I see. Thanks @tapi! I wonder if this change would work with CocoaPods `0.35` stable. 

We don't want to make this change only for developers who use the pre-release version and break the compatibility to the stable version. 

We should make sure that this change works with both the stable and pre-release versions of CocoaPods.

**Comment by steilerDev** [comment]
In your project there is no dynamic library install name (I think this is due to the fact that you are using static libraries within Obj-C)
![screen shot 2015-03-02 at 17 17 37](https://cloud.githubusercontent.com/assets/7031616/6444815/536b11ec-c100-11e4-9dcd-d48b96017b6a.png)
Within my project a dynamic library name is automatically given by cocoapods, so it should be because of your pop configuration. The name there is `_1PasswordExtension`
![screen shot 2015-03-02 at 17 20 09](https://cloud.githubusercontent.com/assets/7031616/6444839/8de64c74-c100-11e4-92a4-92cfcd0868ff.png)

**Comment by tapi** [comment]
It shouldn't affect previous versions of Cocoapods since the podspec is basically just a ruby dict that Cocoapods reads. It doesn't care about the presence of extra keys just the absence of required ones.

**Comment by steilerDev** [comment]
@radazzouz: If you are concerned about compatibility: A note within the readme would be a great start. I don't have a problem of importing `_1PasswordExtension`, but it was hard finding out that I had to use this syntax.

**Comment by radazzouz** [comment]
@steilerDev we want to fix it properly. If the fix consists of adding `module_name = '1PasswordExtension'` to the `Podspec` file, I think that we should do it.

**Comment by radazzouz** [comment]
I don't think that `module_name = '1PasswordExtension'` is the right fix (yet):

* I just created the [bugfix/added-compatibility-for-cocoaPods-0.36](https://github.com/AgileBits/onepassword-app-extension/tree/bugfix/added-compatibility-for-cocoaPods-0.36) branch.

* Updated to `CocoaPods 0.36.0.rc.1`
* Updated my project's `Podfile` as follows:

```
pod '1PasswordExtension', :git => 'https://github.com/AgileBits/onepassword-app-extension.git', :commit => '1489f41115eeb9e381ac652fa656aceb9f3d515b'
```

And I am getting this:

![](https://www.evernote.com/shard/s340/sh/bedfd6a4-dcbc-4d62-9738-9332b42f8fdd/2cb99715d2c3d7c6d3cb5eef4877eeb2/deep/0/ACME-CocoaPods---bash---103-23.png)

Can anyone else confirm this?

Thanks!

**Comment by radazzouz** [comment]
@steilerDev and @tapi , according [Naming Your Product Module](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/BuildingCocoaApps/MixandMatch.html) from this Apple documentation page:

>  By default, your product module name is the same as your product name. However, if your product name has any non-alphanumeric characters, such as a period (.), they are replaced with an underscore (_) in your product module name.

The question is that why is this happening since our product module name starts with `1` which is an alphanumeric character?

Any thoughts?

Thanks!

**Comment by tapi** [comment]
> If the name begins with a number, the first number is replaced with an underscore

Think thats your answer right there ;)

**Comment by radazzouz** [comment]
@tapi so basically we need to create a new CocoaPod named `OnePasswordExtension`, right?

**Comment by tapi** [comment]
@radazzouz may be able to set the header_dir in the podspec its similar to the module_name discussed earlier

**Comment by radazzouz** [comment]
@tapi I did try both these changes without any success. see my latest commit [https://github.com/AgileBits/onepassword-app-extension/commit/4dc926faa572405a8d45d918cde00e1d81d453b7](4dc926faa572405a8d45d918cde00e1d81d453b7).

**Comment by radazzouz** [comment]
Hey guys,

I think finally I was able to figure it out :smile: 

We didn't need to add `module_name`. We only needed to add `header_dir`. 

The good news is that it seems to work with both the stable (0.35) and the pre-release (0.36) versions of CocoaPods.

Here's what I used my `Podfile`:

```
pod '1PasswordExtension', :git => 'https://github.com/AgileBits/onepassword-app-extension.git', :branch => 'bugfix/added-compatibility-for-cocoaPods-0.36'
```

Can you please confirm that this change works properly? If all is good I'll PR the change shortly.

Best,

**Comment by tapi** [comment]
Just confirmed that this fix works.

@steilerDev If your not already you can add `use_frameworks!` to your Podfile to build it as a framework which allows you to import it from swift directly and bypass the bridging header.

**Comment by steilerDev** [comment]
@tapi & @radazzouz: Looks very good! Thanks for your effort. Do you know when you are going to include this change into the main release?

**Comment by radazzouz** [comment]
Hi @steilerDev,

Thanks for confirming the fix :+1: 

I just merged PR #182, so the fix is already on master. If you want to us it, please make sure that your podfile references the master branch and not the `1.1.2` release tag.

```
pod '1PasswordExtension', :git => 'https://github.com/AgileBits/onepassword-app-extension.git', :branch => 'master'
```

Cheers!
